Basic mail functions

IReallyHateLogins opened the pull request:
Added basic mail functions

- Get messages from folder
- Send message and save to sent folder
- Move message to trash or remove permanently from trash

Also:
- Added wrapper for internal javax message class.
- Added endpoints for testing

Commit:
Started to send mail

Commit:
Merge branch 'temp-mail-branch' into get-mail-list-function

Commit:
Added class PostMessage to store message. Rewrite function getMailList in IMAPConnector

Commit:
Added function to send email

Function that sends email and saves it thru imap. Also changed mail wrapper to use strings instead of internal javax classes.

Commit:
Added send and delete function

Also added some endpoints for testing

ValentinBodrov committed:
Merge branch 'develop' into get-mail-list-function

ValentinBodrov reviewed (dismissed):
### This is great
In future we'll format these code line, but it looks pretty now

ValentinBodrov committed:
Fixed inaccuracies in resolving conflicts

ValentinBodrov reviewed (approved):
### Fixed duplicate constructors

Commit:
Upgrade PostMessage class.

Added fields numberInFolder, recipients folder.
Added correctly return of multipart content.

snowzhop reviewed (approved):
Looks like working server
